ts/src/xt.ts

```diff
@@ -0,0 +1,4239 @@
+        //
+        // spot: fetchTrades
+        //
+        //     {
+        //         "i": 203530723141917063,
+        //         "t": 1678227505815,
+        //         "p": "22038.81",
+        //         "q": "0.000978",
+        //         "v": "21.55395618",
+        //         "b": true
+        //     }
+        //
+        // swap and future: fetchTrades
+        //
+        //     {
+        //         "t": 1678227683897,
+        //         "s": "btc_usdt",
+        //         "p": "22031",
+        //         "a": "1067",
+        //         "m": "BID"
+        //     }
+        //
+        // spot: fetchMyTrades
+        //
+        //     {
+        //         "symbol": "btc_usdt",
+        //         "tradeId": "206906233569974658",
+        //         "orderId": "206906233178463488",
+        //         "orderSide": "SELL",
+        //         "orderType": "MARKET",
+        //         "bizType": "SPOT",
+        //         "time": 1679032290215,
+        //         "price": "25703.46",
+        //         "quantity": "0.000099",
+        //         "quoteQty": "2.54464254",
+        //         "baseCurrency": "btc",
+        //         "quoteCurrency": "usdt",
+        //         "fee": "0.00508929",
+        //         "feeCurrency": "usdt",
+        //         "takerMaker": "TAKER"
+        //     }
+        //
+        // swap and future: fetchMyTrades
+        //
+        //     {
+        //         "orderId": "207260566170987200",
+        //         "execId": "207260566790603265",
+        //         "symbol": "btc_usdt",
+        //         "quantity": "13",
+        //         "price": "27368",
+        //         "fee": "0.02134704",
+        //         "feeCoin": "usdt",
+        //         "timestamp": 1679116769838,
+        //         "takerMaker": "TAKER"
+        //     }
+        //
+        const marketId = this.safeString2 (trade, 's', 'symbol');
+        let marketType = (market !== undefined) ? market['type'] : undefined;
+        if (marketType === undefined) {
+            marketType = ('b' in trade) || ('bizType' in trade) ? 'spot' : 'contract';
+        }
+        market = this.safeMarket (marketId, market, '_', marketType);
+        const bidOrAsk = this.safeString (trade, 'm');
+        let side = this.safeStringLower (trade, 'orderSide');
+        if (bidOrAsk !== undefined) {
+            side = (bidOrAsk === 'BID') ? 'buy' : 'sell';
+        }
+        const buyerMaker = this.safeValue (trade, 'b');
+        let takerOrMaker = this.safeStringLower (trade, 'takerMaker');
+        if (buyerMaker !== undefined) {
+            takerOrMaker = buyerMaker ? 'maker' : 'taker';
+        }
+        const timestamp = this.safeIntegerN (trade, [ 't', 'time', 'timestamp' ]);
+        const quantity = this.safeString2 (trade, 'q', 'quantity');
+        const amount = (marketType === 'spot') ? quantity : Precise.stringMul (quantity, this.numberToString (market['contractSize']));
+        return {
```

Comment:
we're missing `safeTrade` here and by using it we can provide all values as strings